### PR TITLE
SASS should write a warning if a regular .scss file is used  when useCSSModules=true

### PR DESCRIPTION
**Pull request by nickpape-msft**
SASS should write a warning if a regular .scss file is used  when useCSSModules=true

### CODE REVIEW

File: gulp-core-build-sass/src/SassTask.ts [hunk 1]
```diff
@@ -76,7 +78,13 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
 
     if (this.taskConfig.useCSSModules) {
       this.logVerbose('Generating css modules.');
-      return this._processFiles(gulp, srcPattern, completeCallback, modulePostCssPlugins);
+      return this._processFiles(gulp, srcPattern, completeCallback, modulePostCssPlugins,
+        (file: gulpUtil.File) => {
+          if (!path.basename(file.path).match(/module\.css$/)) {
```
Comment (dzearing) [inline gulp-core-build-sass/src/SassTask.ts]: Isnt this match wrong? Shouldnt it be scss?

But also why? Why would this filename format matter? Do we treat scss vs module.scss files differently?

Comment (nickpape-msft) [inline gulp-core-build-sass/src/SassTask.ts]: Yes, we do treat module.scss and scss files differently... And this warning is only written if we have "shouldUseModuleCss" turned on, but then don't use module.scss. I can introduce a flag to turn this on or off, if you'd like.

File: gulp-core-build-sass/src/SassTask.ts [hunk 2]
```diff
@@ -41,8 +47,9 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
     sassMatch: [
       'src/**/*.scss'
     ],
-    useCSSModules: false,
-    dropCssFiles: false
+    treatAllFilesAsCSSModules: false,
```
Comment (dzearing) [inline gulp-core-build-sass/src/SassTask.ts]: shouldn't we just keep the `useCSSModules` setting name the same so nothing breaks?

File: gulp-core-build-sass/src/SassTask.ts [hunk 3]
```diff
@@ -16,18 +18,22 @@ export interface ISassTaskConfig {
   postamble?: string;
   /** An array of glob patterns for locating SASS files. */
   sassMatch?: string[];
-  /** 
-   * If this option is specified, files ending with .module.scss extension will
+  /**
+   * If this option is specified, ALL files will be treated as a module.scss and will
```
Comment (dzearing) [inline gulp-core-build-sass/src/SassTask.ts]: but now i understand why you misunderstood! bad comment!

File: gulp-core-build-sass/README.md [hunk 4]
```diff
@@ -27,10 +27,10 @@ An array of glob patterns for locating SASS files.
 **Default:** `['src/**/*.scss']`
```
Comment (dzearing) [inline gulp-core-build-sass/README.md]: don't forget to add an npmx change file. :)

File: gulp-core-build-sass/src/SassTask.ts (hunk 2 above)
Comment (nickpape-msft) [inline gulp-core-build-sass/src/SassTask.ts]: Perhaps... I was on the fence about this. I had apparently misunderstood the meaning of this flag. I thought it had to be on to support css modules, but really it treats EVERYTHING as a css module, so I think it should be renamed.

We aren't using this flag anywhere. Are you?

File: gulp-core-build-sass/src/SassTask.ts (hunk 3 above)
Comment (nickpape-msft) [inline gulp-core-build-sass/src/SassTask.ts]: Right, but we do this anyway as long as the file is named .module.css, regardless of whether this flag is on or not.

File: gulp-core-build-sass/README.md (hunk 4 above)
Comment (nickpape-msft) [inline gulp-core-build-sass/README.md]: Will do

File: gulp-core-build-sass/src/SassTask.ts (hunk 2 above)
Comment (nickpape-msft) [inline gulp-core-build-sass/src/SassTask.ts]: I opened an issue for this, but will revert the change for now

File: gulp-core-build-sass/src/SassTask.ts [hunk 5]
```diff
@@ -72,15 +79,26 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
...
+
+    const checkFilenameForCSSModule: (file: gulpUtil.File) => void = (file: gulpUtil.File) => {
+      if (!path.basename(file.path).match(/module\.scss$/)) {
+
+        const filepath: string = path.relative(this.buildConfig.rootPath, file.path);
+        this.logWarning(`${filepath}: filename should end with module.scss`);
+      }
+    };
+
+    console.log('warnOnNonCSSModules: ' + this.taskConfig.warnOnNonCSSModules);
```
Comment (dzearing) [inline gulp-core-build-sass/src/SassTask.ts]: Should you remove this console.log?

You can use `this.logVerbose` I believe if you need to log verbose stuff.
